rootfs/ingest/python/qserv/jsonparser.py

```diff
@@ -161,9 +163,33 @@ def get_chunk_location(responseJson: dict) -> Tuple[str, int]:
     """Retrieve chunk location (worker host and port)
     inside json response issued by replication service
     """
-    host = responseJson["location"]["http_host"]
+    fqdns = responseJson["location"]["http_host_name"]
     port = int(responseJson["location"]["http_port"])
-    return (host, port)
+    fqdn = get_fqdn(fqdns, port)
+    if not fqdn:
+        raise IngestError(f"Unable to find a valid worker fqdn in json response {responseJson}")
+    return (fqdn, port)
+
+
+def get_fqdn(fqdns: str, port: int, scheme: str = "http") -> str:
+    """Return fqdn of the first reachable scheme://fqdn:port entry
+
+    Parameters
+    ----------
+    fqdns: `str`
+        comma-separated list of fqdns
+    port: `int`
+        url port to reach
+
+    Returns:
+        str: first reachable host fqdn, empty string if not fqdn is reachable
```

Comment:
This should read:

    Returns
    -------
    fdqn : `str`
        First reachable host fqdn, empty string if not fqdn is reachable.